Tests/FeedStoreIntegrationTests.swift

```diff
@@ -27,60 +27,73 @@ class FeedStoreIntegrationTests: XCTestCase {
 	}
 	
 	func test_retrieve_deliversEmptyOnEmptyCache() throws {
-//		let sut = try makeSUT()
-//
-//		expect(sut, toRetrieve: .empty)
+		let sut = try makeSUT()
+
+		expect(sut, toRetrieve: .empty)
 	}
 	
 	func test_retrieve_deliversFeedInsertedOnAnotherInstance() throws {
-//		let storeToInsert = try makeSUT()
-//		let storeToLoad = try makeSUT()
-//		let feed = uniqueImageFeed()
-//		let timestamp = Date()
-//
-//		insert((feed, timestamp), to: storeToInsert)
-//
-//		expect(storeToLoad, toRetrieve: .found(feed: feed, timestamp: timestamp))
+		let storeToInsert = try makeSUT()
+		let storeToLoad = try makeSUT()
+		let feed = uniqueImageFeed()
+		let timestamp = Date()
+
+		insert((feed, timestamp), to: storeToInsert)
+
+		expect(storeToLoad, toRetrieve: .found(feed: feed, timestamp: timestamp))
 	}
 	
 	func test_insert_overridesFeedInsertedOnAnotherInstance() throws {
-//		let storeToInsert = try makeSUT()
-//		let storeToOverride = try makeSUT()
-//		let storeToLoad = try makeSUT()
-//
-//		insert((uniqueImageFeed(), Date()), to: storeToInsert)
-//
-//		let latestFeed = uniqueImageFeed()
-//		let latestTimestamp = Date()
-//		insert((latestFeed, latestTimestamp), to: storeToOverride)
-//
-//		expect(storeToLoad, toRetrieve: .found(feed: latestFeed, timestamp: latestTimestamp))
+		let storeToInsert = try makeSUT()
+		let storeToOverride = try makeSUT()
+		let storeToLoad = try makeSUT()
+
+		insert((uniqueImageFeed(), Date()), to: storeToInsert)
+
+		let latestFeed = uniqueImageFeed()
+		let latestTimestamp = Date()
+		insert((latestFeed, latestTimestamp), to: storeToOverride)
+
+		expect(storeToLoad, toRetrieve: .found(feed: latestFeed, timestamp: latestTimestamp))
 	}
 	
 	func test_delete_deletesFeedInsertedOnAnotherInstance() throws {
-//		let storeToInsert = try makeSUT()
-//		let storeToDelete = try makeSUT()
-//		let storeToLoad = try makeSUT()
-//
-//		insert((uniqueImageFeed(), Date()), to: storeToInsert)
-//
-//		deleteCache(from: storeToDelete)
-//
-//		expect(storeToLoad, toRetrieve: .empty)
+		let storeToInsert = try makeSUT()
+		let storeToDelete = try makeSUT()
+		let storeToLoad = try makeSUT()
+
+		insert((uniqueImageFeed(), Date()), to: storeToInsert)
+
+		deleteCache(from: storeToDelete)
+
+		expect(storeToLoad, toRetrieve: .empty)
 	}
 	
 	// - MARK: Helpers
 	
 	private func makeSUT() throws -> FeedStore {
-		fatalError("Must be implemented")
+		let storeURL = testSpecificStoreUrl()
+		return try CoreDataFeedStore(storeURL: storeURL)
+	}
+	
+	private func testSpecificStoreUrl() -> URL {
+		return cachesDirectory().appendingPathComponent("\(type(of: self)).store")
+	}
+	
+	private func cachesDirectory() -> URL {
+		return FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
 	}
 	
 	private func setupEmptyStoreState() throws {
-		
+		deleteStoreArtifacts()
+	}
+	
+	private func deleteStoreArtifacts() {
+		try? FileManager.default.removeItem(at: testSpecificStoreUrl())
```

Comment:
Instead of ignoring errors, this method can throw. If there's something wrong, it'll generate a test failure. This way, it's easier to find and solve issues.

```suggestion
	private func deleteStoreArtifacts() throws {
		try FileManager.default.removeItem(at: testSpecificStoreUrl())
```

Reply:
Agree, it's also needed to check if artifacts exist or not